oneclick.py

```diff
@@ -0,0 +1,600 @@
+from nose.tools import set_trace
+from collections import defaultdict
+import datetime
+import base64
+import os
+import json
+import logging
+import re
+
+from config import (
+    Configuration, 
+    temp_config,
+)
+
+from util import LanguageCodes
+#from util.xmlparser import XMLParser
+#from util.jsonparser import JsonParser
+from util.http import (
+    HTTP,
+    #RemoteIntegrationException,
+)
+from coverage import CoverageFailure
+
+from model import (
+    Contributor,
+    DataSource,
+    DeliveryMechanism,
+    Edition,
+    Hyperlink,
+    Identifier,
+    Representation,
+    Subject,
+)
+
+from metadata_layer import (
+    CirculationData,
+    ContributorData,
+    FormatData,
+    IdentifierData,
+    LinkData,
+    Metadata,
+    SubjectData,
+)
+
+from config import Configuration
+from coverage import BibliographicCoverageProvider
+
+class OneClickAPI(object):
+
+    API_VERSION = "v1"
+    #DATE_FORMAT = "%m-%d-%Y"
+
+    # a complete response returns the json structure with more data fields than a basic response does
+    RESPONSE_VERBOSITY = {0:'basic', 1:'compact', 2:'complete', 3:'extended', 4:'hypermedia'}
+
+    log = logging.getLogger("OneClick API")
+
+    def __init__(self, _db, library_id=None, username=None, password=None, 
+        remote_stage=None, base_url=None, basic_token=None):
+        self._db = _db
+        (env_library_id, env_username, env_password, 
+         env_remote_stage, env_base_url, env_basic_token) = self.environment_values()
```

Comment:
I think when I gave you this advice I was looking at my refactored millenium_patron.py, which hasn't landed yet. Leave this alone for now; it's not vital and we can change it later.